app/helpers/publishers_helper.rb

```diff
@@ -418,4 +418,14 @@ def youtube_login_permitted?(channel)
   def publisher_created_through_youtube_auth?(publisher)
     publisher && publisher.channels.visible.count == 1 && youtube_login_permitted?(publisher.channels.visible.first)
   end
+
+  def emailed_auth_token_action
```

Comment:
The purpose of this method is to determine whether the text should say
"we've sent you a log in link"
or
"we've sent you a sign up link"

we could remove this method if didn't care about differentiating, and just said
"we've sent you an access link"

Reply:
I just tested this with both an existing and a new email address, and both displayed "We just sent a sign up link to ..." in emailed_auth_token.html.slim, despite receiving a login email for one and a signup email for the other. I'll investigate what's happening in this helper.

Reply:
Ah, this happened because the session was not cleared before performing the login and still contained `session[:created_publisher_id]`, which is what the helper is keying off of.

Reply:
Thanks.  I think this method may be more trouble than it's worth.

This method originally used `params[:action]` to determine what to display there.  If the action was `create_done` it used "log in" and if it was `create_auth_token` it used "sign up".  However, if they click "send again" at that point, the action that loads this view is `resend_email_verify_email`, and you can no longer differentiate.  That's why I tried grabbing the user from the session.

If you have a good way to choose whether to display "log in" vs "sign up" i'll use that.  Otherwise I'll remove this method and I'll just say "access link".

Reply:
@nvonpentz perhaps the mode could be passed in from the controller?

```
render(:emailed_auth_token, locals: { mode: :signup })
```

and 

```
render(:emailed_auth_token, locals: { mode: :login })
```

And then the helper could be passed the `mode` instead of relying on a session var?

Reply:
@dgeb Hmm, I think we would run in the same problem as checking the `params[:action]`.  That is, when we click "send email again" it will always display "sign up" since `resend_email_verify_email` redirects to `create_done` which will pass the `:sign_up` mode.

https://github.com/brave-intl/publishers/blob/aa69e380817f4e612511c8a59b94d7fbcc4b14a1/app/controllers/publishers_controller.rb#L100

Reply:
@nvonpentz I see. Yeah, the `mode` would have to be passed back in that "send email again" request to complete the cycle. Alternatively, you could clear `session[:created_publisher_id]` in `PublishersController#create_auth_token` before rendering and continue to use your current helper.